### ISSUE TITLE
Is this project alive ?

### ISSUE THREAD

**Issue by mackwic** [vispy/numpy.js#1]
Hi @rossant. Just checking in if this project will be continued ...?

**Comment by rossant** [comment]
Hi @mackwic. It hasn't really started yet actually, because it's not on the top of our list currently. We'd be happy to help anyone interested though. Would you have an interest in this project?

**Comment by mackwic** [comment]
Yes. A lot. I would like to have numpy in nw.js for our data miners. :)
What's your estimated timeline ? I may contribute in near future (~1 month).

**Comment by rossant** [comment]
We don't have plans to work ourselves directly on this in the near future, but we're looking forward to getting help from external contributors (including potential GSoC students).

If you want a bit of context, I suggest you to read the following blog posts:
- http://cyrille.rossant.net/scientific-python-in-the-browser-its-coming/
- http://cyrille.rossant.net/numpy-browser-llvm/

In the second post you'll see some very early steps with a different approach to get NumPy in the browser. I think both approaches are doable but require quite some work.

**Comment by mackwic** [comment]
In my specific case, I want to get rid of python, so I would tend for numpy.js as pure JS (eventually with C++ bindings for a node.js version).

But, this is a gigantic project to do alone. I was just checking in and inform that I would gladly help. Also, glad to see there is still some love here. ;)

**Comment by rossant** [comment]
I would be delighted to get some help :) This is a big project but not "that" big IMO. With an adequate generic interface for universal functions it would be easy to implement the most useful functionality of NumPy, that is vectorized operations.

I'm not sure if you're aware of [this](https://github.com/scijs/ndarray), I haven't looked at it in details yet, but it could be an interesting approach.

**Comment by mackwic** [comment]
Yeah, ndarray seems promising.

The thing is, only a third of the work will be on the implementation. You have to take a third on the test suite and documentation of the caveats, and another third will be benchmarking and optimization (this third should be disposable in the short/middle term).
Don't underestimate the importance of the testing effort. It's even more important than the isofunctionnality with numpy IMHO. The profiling/optimization effort is simply insane at short term, so it's simply a no-go.

**Comment by rossant** [comment]
Right. Personally I'm less interested in the performance than having something that just works, even if it's not blazingly fast. For highest performance you might need some code generation for ufuncs.

**Comment by almarklein** [comment]
Is anyone (planning to) working on this? We want something like this for Bokeh too: https://github.com/bokeh/bokeh/issues/2209

**Comment by rossant** [comment]
I probably wont have time to be the main developer, but i would be happy to contribute
